### ISSUE TITLE
Missing images are not handled

### ISSUE THREAD

**Issue by kxt** [hipstermojo/paperoni#11]
Hey, love your project!

With one site I was testing it, I noticed that the resulting epub contains an img src reference to the error 404 status page.

Reproduction is trivial:
* try a page with `<img src="nonexistent"/>`
* resulting epub will have an `OEBPS/index.html` entry like this: `<img src="a2dafe1d007e2607f549726f0e7c9886.html"></img>` and an `OEBPS/a2dafe1d007e2607f549726f0e7c9886.html` containing the error 404 page.

Probably the html fetching code in `fetch_html` in `http.rs` should be extracted into a helper function and used in `download_images` too, as it handles both error status codes and redirection.

Not sure what would be the right course of action for when the image is missing, perhaps inserting the original (broken) url as source, so if the problem is transient and the image url starts to work, readers may load it then.

PS: this bug is only a manifestation of another bug I'm still hunting, looks as if the image url gets mangled somehow and that's what causes the 404. I'll open another bug as soon as I figured out what's happening.

**Comment by hipstermojo** [comment]
Hi @kxt, thanks for trying out Paperoni and also investigating the issue you faced. Might I ask for the URL of the website you were testing it on and the version of Paperoni you used on it?
The current version of Paperoni `0.4.0-alpha1` has redirects enabled for downloading images as well as leaving the original URL in place in case it is unable to fetch it. In case you are using the latest version then I will look to fix this bug before the next release which should be on May 24th.

**Comment by kxt** [comment]
I'm using 0.4.0-alpha1.

I've noticed the bug with multiple pages on substack, eg this one: https://blindsquirrel.substack.com/p/etsy-the-power-of-special . But this already involves the other bug mentioned where the image url gets rewritten to an invalid one, and I haven't yet figured out that one.

To have a clean reproduction, I just created a bogus html file like this:
```
<html>
<head><title>Test page</title></head>
<body>Hello world! <img src="missing.png"/></body>
</html>
```
And then served it locally with python: `python -mhttp.server`

The problem is that this is not a fetch error that the `match req.await` catches. The error 404 page is a successful fetch containing a content type of text/html, so the error code should be checked.
